Controller/Index.php

```diff
@@ -113,4 +116,25 @@ protected function writeLog()
         $this->altapayLogger->addDebugLog('-- Params --', $this->getRequest()->getParams());
         $this->altapayLogger->addDebugLog('- END', $calledClass);
     }
+
+    /**
+     * @param string $orderId
+     *
+     * @return mixed
+     */
+    protected function setSuccessPath($orderId)
+    {
+        $resultRedirect = $this->redirectFactory->create();
+        if ($orderId) {
+            $order = $this->order->loadByIncrementId($orderId);
+            $hashOrderID = hash("sha256", $orderId);
```

Comment:
The code is public, meaning that the attacker can guess the hashes by looking at the code, and try multiple orderIds. You should use a secret (salt) for hashing that is configurable.

Reply:
@emicha Updated the code to use Magento's default random hash mechanism instead of calculating the hash of order ID. Also, we are removing the hash reference from DB once the customer is redirected to the success page.